platform-tests/example-apps/healthcheck/db.go

```diff
@@ -105,3 +129,94 @@ func mysqlOpen(dbu string, ssl bool) (*sql.DB, error) {
 
 	return sql.Open("mysql", connString)
 }
+
+func testMongoDBConnection(session *mgo.Session) error {
+	defer session.Close()
+
+	type Person struct {
+		Name  string
+		Phone string
+	}
+
+	input := &Person{Name: "John Jones", Phone: "+447777777777"}
+	db := session.DB("test").C("people")
+	err := db.Insert(input)
+	if err != nil {
+		return err
+	}
+
+	var result Person
+	err = db.Find(bson.M{"name": "John Jones"}).One(&result)
+	if err != nil {
+		return err
+	}
+	if result.Name != input.Name {
+		return fmt.Errorf("Name unexpectedly changed to %s in MongoDB test.", result.Name)
+	}
+	if result.Phone != input.Phone {
+		return fmt.Errorf("Phone unexpectedly changed to %s in MongoDB test.", result.Phone)
+	}
+	return nil
+}
+
+func mongoDBOpen(vcap_services_str string, ssl bool) (*mgo.Session, error) {
+	type VCAPService struct {
+		Credentials struct {
+			URI                 string `json:"uri"`
+			CACertificateBase64 string `json:"ca_certificate_base64"`
+		} `json:"credentials"`
+	}
+
+	type VCAPServices struct {
+		MongoDB []VCAPService `json:"mongodb"`
+	}
+
+	// Retrieve the MongoDB connection URI and the CA Certificate from the `VCAP_SERVICES`
+	// environment variable.
+	var vcap_services VCAPServices
+	if err := json.Unmarshal([]byte(vcap_services_str), &vcap_services); err != nil {
+		return nil, err
+	}
+	mongodb_credentials := vcap_services.MongoDB[0].Credentials
+	dbu := mongodb_credentials.URI
+	ca_certificate_base64 := mongodb_credentials.CACertificateBase64
+
+	// This is work around for https://github.com/go-mgo/mgo/issues/84
+	uri := strings.TrimSuffix(dbu, "?ssl=true")
+	mongourl, err := mgo.ParseURL(uri)
+	if err != nil {
+		return nil, err
+	}
+
+	if !ssl {
+		session, err := mgo.DialWithTimeout(uri, 5 * time.Second)
+		if err != nil {
+			return nil, err
+		}
+		return session, nil
+	}
+
+	// Compose has self-signed certs for mongo. Make sure we verify it against CA certificate brovided in binding
```

Comment:
brovided
    _conjunction_

    provided by a bro

I'm adding this to Urban Dictionary.